### PR TITLE
Fix dependencies, leading to compile time warnings

### PR DESCRIPTION
**Pull request by cheenamalhotra**
"Microsoft.SqlServer.Management.SmoMetadataProvider" dependency is required in SqlTools and Kusto Service layers, but is referenced through Managed batch parser, which leads to below compile time warning:

`~\src\Microsoft.SqlTools.ManagedBatchParser\Microsoft.SqlTools.ManagedBatchParser.csproj : warning NU1701: Package 'Microsoft.SqlServer.Management.SmoMetadataProvider 170.7.0-preview' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETStandard,Version=v2.1'. This package may not be fully compatible with your project.`

"Microsoft.SqlServer.DacFx" dependency is unwanted in Managed Batch Parser, also leading to similar compiler warning:

`~\src\Microsoft.SqlTools.ManagedBatchParser\Microsoft.SqlTools.ManagedBatchParser.csproj : warning NU1701: Package 'Microsoft.SqlServer.DacFx 160.6295.0-preview' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. `

### CODE REVIEW

File: src/Microsoft.Kusto.ServiceLayer/Microsoft.Kusto.ServiceLayer.csproj [hunk 1]
```diff
@@ -32,6 +32,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.OperationalInsights" />
     <PackageReference Include="Microsoft.SqlServer.DACFx" />
+	<PackageReference Include="Microsoft.SqlServer.Management.SmoMetadataProvider" />
```
Comment (Charles-Gagnon) [inline src/Microsoft.Kusto.ServiceLayer/Microsoft.Kusto.ServiceLayer.csproj:35]: Fix spacing

File: src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj [hunk 2]
```diff
@@ -12,9 +12,7 @@
 	</ItemGroup>
 	<ItemGroup>
 		<PackageReference Include="Microsoft.Data.SqlClient" />
-		<PackageReference Include="Microsoft.SqlServer.DACFx" />
 		<PackageReference Include="Microsoft.SqlServer.SqlManagementObjects" />
-		<PackageReference Include="Microsoft.SqlServer.Management.SmoMetadataProvider" />
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj:17]: Hm, I don't suppose you know why these were here in the first place?

**Review by Charles-Gagnon — APPROVED** [review]
I guess if it's compiling then there shouldn't be any runtime issues barring some silly reflection nonsense (which I don't think we do...)

I'd sync up with Matteo and make sure he isn't relying on these references for his powershell stuff before getting this in though.

File: src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj (hunk 2 above)
Comment (cheenamalhotra) [reply src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj:17]: There doesn't seem to be an obvious reason. If dependencies need them, they can import themselves.

**Review by alanrenmsft — APPROVED** [review]

**Review by Matteo-T — APPROVED** [review]
:shipit:

**Comment by Matteo-T** [comment]
Thanks for pinging me. I think that the *removal* of stuff (particularly unused references) is always a good thing. :)

---
In reply to: [1122840032](https://github.com/microsoft/sqltoolsservice/pull/1713#issuecomment-1122840032) [](http://example.com/codeflow?ancestors=1122840032,1122840032)

**Comment by Matteo-T** [comment]
Will you kindly let me know when a new official (signed) build of this is available and where I can download it from (not super urgent, so take your time). Thanks

---
In reply to: [1261698756](https://github.com/microsoft/sqltoolsservice/pull/1713#issuecomment-1261698756) [](http://example.com/codeflow?ancestors=1261698756,1122840032,1122840032)

**Pull request merged**